server/routerlicious/packages/services/src/mongodb.ts

```diff
@@ -204,23 +206,44 @@ export class MongoCollection<T> implements core.ICollection<T>, core.IRetryAble
     }
 
     private async requestWithRetry<TOut>(request: () => Promise<TOut>, callerName: string): Promise<TOut> {
-        return requestWithRetry<TOut>(
-            request,
-            callerName,
-            {}, // telemetryProperties
-            (e) => this.retryEnabled && MongoErrorRetryAnalyzer.shouldRetry(e), // ShouldRetry
-            3, // maxRetries
-            1000, // retryAfterMs
-            (error: any, numRetries: number, retryAfterInterval: number) =>
-                numRetries * retryAfterInterval, // retryAfterIntervalCalculator
-        );
+        let metric: Lumber<LumberEventName.MongoQuery>;
+        if (this.telemetryEnabled) {
+            metric = Lumberjack.newLumberMetric(LumberEventName.MongoQuery);
+            metric.setProperty("callerName", callerName);
```

Comment:
@pradeepvairamani  I added configurable telemetry here

Reply:
This custom `requestWithRetry` here calls the main [`requestWithRetry`](https://github.com/microsoft/FluidFramework/blob/ea4f1177457214f66e2b30364f1c9a412a18d1e6/server/routerlicious/packages/services-core/src/runWithRetry.ts#L95) on line 216. The main `requestWithRetry` already includes some telemetry in there. Was that not enough? If instead we wanted a metric (the main `requestWithRetry` only logs), do we want to instead update the main one to also use metrics? The way we are doing it here seems a bit of a cyclical